chore: cleanup Platform FAQs section (#17809)
calcom/cal.com: docs/platform/faq.mdx

- An atom is a customizable UI component that handles scheduling on behalf of your user.
- Everything from the front-end to API calls is being handled by the atom, all you need to is import the atom and drop it in your code.

### 6. What are the minimun setup requirments for using atoms?
- You need to have a project that uses React version 18 and above.

> - At the moment atoms are only supported in React, so there's no way to use them in vanilla js or any other popular framework.

Author: I removed the 2nd bullet point because it is not entirely true. Frameworks like Astro support React, and thus should implicitly support atoms as well. Happy to revert this change if you have proof otherwise.

> At present, atoms are written in React, so your project need to use React version 18 or above.

Reviewer: ```suggestion
 You need to have a project that uses React version 18 and above.
 At the moment atoms are only supported in React.
```

this looks much clean and easy to understand.

Author: The second sentence implies this only works in React. Can you show proof of that? Without that, I won’t commit the suggestion as-is.

Either way, maintainers can make edits so feel free to make the change and merge it as you see fit.

Reviewer 2: Why are we taking out the hyphens? The other numbered elements have them.

In terms of React, yes, our atoms can only be used with React right now.
